tests/func/experiments/test_experiments.py

```diff
@@ -629,7 +629,8 @@ def test_fix_exp_head(tmp_dir, scm, tail):
     head = "HEAD" + tail
     assert head == fix_exp_head(scm, head)
 
-    scm.set_ref(EXEC_BASELINE, "refs/heads/master")
+    rev = "1" * 40
+    scm.set_ref(EXEC_BASELINE, rev)
```

Comment:
The original `set_ref(..., "refs/heads/master")` would result in an invalid git ref file (since we didn't pass a SHA into `set_ref` and also did not specify `symbolic=True`). Previously this was irrelevant since for this test we only needed the ref file for `EXEC_BASELINE` to exist for this test (regardless of valid/invalid ref file contents).

However, with the new `get_ref` functionality to handle annotated tags, having an invalid ref file will break the updated `get_ref` call.